### PR TITLE
DNS Auth Zone Resource and Data Source Implementation

### PR DESCRIPTION
**Pull request by unasra**

### CODE REVIEW

File: examples/data-sources/bloxone_dns_auth_zones/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Get auth zones filtered by an attribute
+data "bloxone_dns_auth_zone" "example_by_attribute" {
+  filters = {
+    "fqdn"         = "tf-acc-test.com."
+    "primary_type" = "cloud"
```
Comment (venkat-iblox) [inline examples/data-sources/bloxone_dns_auth_zones/data-source.tf:5]:
```suggestion
    fqdn        = "tf-acc-test.com."
    primary_type = "cloud"
```

File: examples/data-sources/bloxone_dns_auth_zones/data-source.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  }
+}
+
+# Get auth zones filtered by tag
+data "bloxone_dns_auth_zone" "example_by_tag" {
+  tag_filters = {
+    "region" = "eu"
```
Comment (venkat-iblox) [inline examples/data-sources/bloxone_dns_auth_zones/data-source.tf:12]:
```suggestion
    region = "eu"
```

File: internal/service/dns_config/api_auth_zone_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+)
+
+//TODO: add tests
+// The following require additional resource/data source objects to be supported.
+// - inheritance_sources
+// - ACL Type - TSIG Key
+// - ACL Type - ACL
+// - internal_secondaries
+// - nsgs
+// - zone_authority : Mname and rname provide inconsistent result after apply
```
Comment (venkat-iblox) [inline internal/service/dns_config/api_auth_zone_resource_test.go:25]:
```suggestion
// - zone_authority : Mname and rname provide inconsistent results after apply
```

File: internal/service/dns_config/api_auth_zone_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,798 @@
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAuthZoneBasicConfig("tf-acc-test.com.", "cloud"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAuthZoneExists(context.Background(), resourceName, &v),
+					// TODO: check and validate these
```
Comment (venkat-iblox) [inline internal/service/dns_config/api_auth_zone_resource_test.go:41]: TODO isn't required here
```suggestion
```

File: internal/provider/provider.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package provider
 import (
 	"context"
 	"fmt"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/service/dns_config"
```
Comment (venkat-iblox) [inline internal/provider/provider.go:6]: please fix the Import ordering

**Review by venkat-iblox — COMMENTED** [review]
In addition to the comments, please resolve the conflicts

File: examples/data-sources/bloxone_dns_auth_zones/data-source.tf (hunk 1 above)
Comment (unasra) [reply examples/data-sources/bloxone_dns_auth_zones/data-source.tf]: Resolved

File: internal/service/dns_config/api_auth_zone_resource_test.go (hunk 4 above)
Comment (unasra) [reply internal/service/dns_config/api_auth_zone_resource_test.go]: Resolved

File: internal/provider/provider.go (hunk 5 above)
Comment (unasra) [reply internal/provider/provider.go]: Resolved

File: internal/provider/provider.go [hunk 6]
```diff
@@ -106,14 +107,16 @@ func (p *BloxOneProvider) DataSources(ctx context.Context) []func() datasource.D
...
 
 		infra_provision.NewUIJoinTokenDataSource,
 
 		infra_mgmt.NewHostsDataSource,
 		infra_mgmt.NewServicesDataSource,
+
+		dns_config.NewAuthZoneDataSource,
+		dns_config.NewViewDataSource,
+		dns_config.NewAuthNsgDataSource,
+		dns_config.NewHostDataSource,
```
Comment (venkat-iblox) [inline internal/provider/provider.go:119]: Can you move this back?
@mathewab Moved this in his PR: https://github.com/infobloxopen/terraform-provider-bloxone/pull/30/files#diff-58d6a027753b50994deb7e11e4a99dde423f35844986019bd9cea5e0c94aba22R90-R92. Let us keep it there

Comment (venkat-iblox) [reply internal/provider/provider.go:119]: I'll address this in my PR

**Review by venkat-iblox — APPROVED** [review]

**Pull request merged**